CMakeLists.txt

```diff
@@ -0,0 +1,203 @@
+        "url for all submodules except cnpm: replace rogii-com to boostorg. This is required "
+        "because we use our fork of the superproject but all submodules are set with relative path "
+        "and we don't fork them. After that 'git submodule update --recursive' may be called as usual. "
+        "For details check https://stackoverflow.com/a/42035018 ."
+    )
+endif()
+
+include(
+    "${PROJECT_SOURCE_DIR}/cmake/npm/NastyPackageManager.cmake"
+)
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    NPM_ADD_PACKAGE(
+        NAME
+            gxx_runtime
+        VERSION
+            9.2.1
+        BUILD_NUMBER
+            0
+    )
+    NPM_ADD_PACKAGE(
+        NAME
+            python
+        VERSION
+            3.7.4
+        BUILD_NUMBER
+            0
+    )
+elseif(MSVC)
+    NPM_ADD_PACKAGE(
+        NAME
+            msvs
+        VERSION
+            2019
+        BUILD_NUMBER
+            1
+    )
+    NPM_ADD_PACKAGE(
+        NAME
+            windowssdk
+        VERSION
+            10.0.18362.0
+        BUILD_NUMBER
+            0
+    )
+    NPM_ADD_PACKAGE(
+        NAME
+            python
+        VERSION
+            3.7.4
+        BUILD_NUMBER
+            0
+        TAG
+            sdk14393_vs2015up3
+    )
+endif()
+
+set(
+    REPOSITORY_OS
+    "windows"
+)
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    set(
+        REPOSITORY_OS
+        "linux"
+    )
+endif()
+
+NPM_PREPARE_PACKAGES(
+    DEFAULT_REPOSITORY_URLS
+    "https://cnpm.rogii.com/${REPOSITORY_OS}/public"
+)
+
+get_property(
+    PYTHON_LIB_LOCATION
+    TARGET
+        Python::library
+    PROPERTY
+        IMPORTED_LOCATION
+)
+
+# get python package directory from imported location "/python/lib/python3.7"
+
+message(
+    STATUS
+    "PYTHON_LIB_LOCATION = ${PYTHON_LIB_LOCATION}"
+)
+
+get_filename_component(PYTHON_LIBDIR ${PYTHON_LIB_LOCATION} DIRECTORY)
+message(
+    STATUS
+    "PYTHON_LIBDIR = ${PYTHON_LIBDIR}"
+)
+
+get_filename_component(PYTHON_LIBDIR ${PYTHON_LIBDIR} DIRECTORY)
+
+message(
+    STATUS
+    "PYTHON_LIBDIR = ${PYTHON_LIBDIR}"
+)
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    set(
+        TOOLSET
+        "gcc"
+    )
+    set(
+        USER_CONFIG
+        "starsteer_linux.jam"
+    )
+    add_custom_command(
+        OUTPUT
+            ${boost_SOURCE_DIR}/project-config.jam
+        COMMAND
+            ./bootstrap.sh
+        WORKING_DIRECTORY
+            ${boost_SOURCE_DIR}
+        USES_TERMINAL
+    )
+    add_custom_command(
+        OUTPUT
+            ${CMAKE_INSTALL_PREFIX}/include/boost-1_72/boost/version.hpp
+        COMMAND
+            ${CMAKE_COMMAND} -E env PYTHON_LIBDIR=${PYTHON_LIBDIR} ./b2 install -j8 toolset=${TOOLSET} variant=debug,release link=static threading=multi runtime-link=shared ${ADDRESS_MODEL} --layout=versioned --build-type=minimal --user-config=${USER_CONFIG} python=3.7 --prefix=${CMAKE_INSTALL_PREFIX}
```

Comment:
Too long line. Need to be splitted by rows

Reply:
done